lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -32,6 +32,9 @@ namespace {
     "The larger the number is, the more fine-grained debug "
     "information will be printed"),
     cl::init(0));
+  static cl::opt<bool> NoBigQuery("souper-exhaustive-synthesis-no-big-query",
```

Comment:
let's get some data about this also -- maybe run synthesis with and without it and see how it changes the CPU time used